internal/util/sizeof_test.go

```diff
@@ -0,0 +1,376 @@
+	var mediaLink string = "media-link"
+	var generation int64 = 858734898
+	var metaGeneration int64 = 858734899
+	const storageClass string = "standard"
+	deleted := time.Now()
+	updated := time.Now()
+	const componentCount int64 = 1
+	const contentDisposition string = "my-content-disposition"
+	const customTime string = "my-custom-time"
+	const eventBasedHold bool = true
+	customMetadaField1 := "google-symlink"
+	customMetadaValue1 := "true"
+	customMetadaField2 := "google-xyz-field"
+	customMetadaValue2 := "google-symlink"
+	customMetadataFields := map[string]string{customMetadaField1: customMetadaValue1, customMetadaField2: customMetadaValue2}
+	customMetadataFieldsContentSize := nestedSizeOfString(&customMetadaField1) + nestedSizeOfString(&customMetadaValue1) + nestedSizeOfString(&customMetadaField2) + nestedSizeOfString(&customMetadaValue2)
+	customAcls := []*storagev1.ObjectAccessControl{
+		{
+			Bucket:     "my-bucket",
+			Domain:     "my-domain",
+			Email:      "my-email@my-domain.com",
+			Entity:     "my-entity",
+			EntityId:   "my-entity-id",
+			Etag:       "my-etag",
+			Generation: generation,
+			Id:         "object-id",
+			Kind:       "object-kind",
+			Object:     "my-object",
+			ProjectTeam: &storagev1.ObjectAccessControlProjectTeam{
+				ProjectNumber: "78358753894",
+				Team:          "project-team",
+				ForceSendFields: []string{
+					"field1", "field2", "field3",
+				},
+			},
+			Role:            "my-role",
+			SelfLink:        "",
+			ServerResponse:  googleapi.ServerResponse{},
+			ForceSendFields: []string{},
+			NullFields:      []string{},
+		},
+	}
+
+	o := gcs.Object{
+		Name:               name,
+		ContentType:        contentType,
+		ContentLanguage:    contentLanguage,
+		CacheControl:       cacheControl,
+		Owner:              owner,
+		Size:               100,
+		ContentEncoding:    contentEncoding,
+		MD5:                &md5Value,
+		CRC32C:             &crc32,
+		MediaLink:          mediaLink,
+		Metadata:           customMetadataFields,
+		Generation:         generation,
+		MetaGeneration:     metaGeneration,
+		StorageClass:       storageClass,
+		Deleted:            deleted,
+		Updated:            updated,
+		ComponentCount:     componentCount,
+		ContentDisposition: contentDisposition,
+		CustomTime:         customTime,
+		EventBasedHold:     eventBasedHold,
+		Acl:                customAcls,
+	}
+
+	var expectedSize int = sizeOfEmptyGcsObject
+	expectedSize += len(contentType) + len(name) + len(contentLanguage) + len(cacheControl) + len(contentEncoding) + len(owner) + len(mediaLink) + len(storageClass) + len(contentDisposition) + len(customTime)
+	expectedSize += md5.Size // for MD5 [md5.Size]byte
+	expectedSize += 4        // for CRC 32 uint32
+	expectedSize += customMetadataFieldsContentSize
+	expectedSize += nestedContentSizeOfArrayOfAclPointers(&customAcls)
+
+	AssertEq(expectedSize, NestedSizeOfGcsObject(&o))
+}
+
+func (t *SizeofTest) TestNestedSizeOfMinObject() {
```

Comment:
We can create a similar test as we did for GCSObject.

Reply:
As NestedSizeOfMinObject isn't used/needed anywhere today, I've removed it altogether. When we switch MinObject, we can switch NestedSizeOfGcsObject itself to NestedSizeOfMinObject.